crates/shared/src/sources/balancer_v2/swap/weighted_math.rs

```diff
@@ -77,6 +77,50 @@ pub fn calc_in_given_out(
     balance_in.mul_up(ratio)
 }
 
+pub fn calc_out_given_in_v3(
```

Comment:
Instead of duplicating these functions I think you could simply have 1 base version that takes the specific `pow_up` function pointer. And then `calc_out_given_in_v3()` would just be a wrapper around the base version with the new `pow_up` function. [example](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=7848d04b7fe2113f105306393bdb1f99)

Reply:
I purposefully kept the functions separate and preferred to copy the small bit of code instead of passing in a function pointer. My rationale is that, from a conceptual level, the difference in how input/output amounts get computed is version-based and it is easy to imagine that it is not just the `pow_up` function that is different but other things (different rounding directions for other operations, etc.). The code that was kept common in `WeighedPool::get_amount_{in,out}` is what is implemented in the `BaseMinimalSwapInfoPool` implementation (so the base class that didn't change).

Reply:
We tried out the suggestion and spoke in person, and agree it is OK as is.